### DISCUSSION TITLE
Uppercase namespaces not working

### DISCUSSION THREAD

**Discussion by PsyLogic** [livewire/livewire#5167] [1. Help] [answered]
Hello,

I encounter an issue while I was trying to create component in sub-folder.

I create a new file `App\Http\Livewire\TEST\TestTable.php`, and `livewire/test/test-table.blade.php`.

The issue when I'm trying to call this view `@livewire('test.test-table')` it will throw Component not Found 'test.test-table.blade'.

To solve this I have to rename the subFolder to  **t-e-s-t** instead of **test**.

How can I map the folder to its component, I tried the discover command but nothing.

Thanks !

Livewire: 2.x

**Comment by joshhanley** [comment]
@PsyLogic I suggest using the Livewire make command to generate the component, as it works for me when I tested it 🙂

When you do, you will see it creates these files:

<img width="709" alt="image" src="https://user-images.githubusercontent.com/882837/183313850-0f7e13af-41db-4070-a818-93c77ff7849a.png">

So it include this in your blade view, you would use 

```blade
<livewire:t-e-s-t.test-table />
```

Livewire converts any `PascalCase` names to `kebab-case` for the view name. So that is why the subdirectory `TEST` becomes `t-e-s-t` but only for the blade folder and the Livewire blade include. This is following Laravel's convention.

Hope this helps!

**Reply by PsyLogic** [reply]
Hello @joshhanley ;

Thank you for your reply.

I tried it with artisan cmd before, but what I want is when Livewire cannot discover things, we should have a class where we can map things.
Sometimes we don't need to follow the conventions and that is why Laravel provides us with Service Providers to map classes.
I thought Livewire can do the same thing

**Comment by PsyLogic** [comment, marked as answer]
Well, I found the answer event it was in the other section of the Doc (package dev).

We can map the component to another alias by using the Livewire::component() method. 
it was mentioned here: 
https://github.com/livewire/livewire/blob/master/src/LivewireManager.php#L64

So to add an Alias, you have to call in on your service provider `boot()`:

```php
class XServiceProvider extends ServiceProvider {
    public function boot() {
        Livewire::component('TEST.test-table', TestTable::class);
    }
}
```

```php
<div>
    @livewire('TEST.test-table')
</div>
```

**Reply by joshhanley** [reply]
@PsyLogic ah yep, sorry didn't understand that's what you were after. Glad you found it 🙂
